### DISCUSSION TITLE
Data is not synced to Firebase when more than one firepad is active

### DISCUSSION THREAD

**Discussion by tyagirajat200** [0xTheProDev/Operational-Transformation#114] [Bug Report]
I have created 3 monaco editors(for HMTL, JAVASCRIPT and CSS) connected with firebase using firepad. But whenever I changed the data in the editor, code is not synced to the firebase.

When I use [@hackerrank/firepad](https://www.npmjs.com/package/@hackerrank/firepad) package, pads are working fine and code is properly synced to the firebase. 

Desktop:
OS: Windows
Browser chrome
Version: 102.0.5005.115 (Official Build) (64-bit)
Framework: Angular 12

**Comment by asoran** [comment]
Hello, it seems that the FirebaseAdapter has some persisting state, so even if you create new instance of it, some stuff persist and the other instances doesn't works.
What I did was very dirty but it works:
Take the lib from node_modules/@otjs/firebase-plaintext/lib/index.js (the minified output version of the FirebaseAdapter), duplicate it somewhere from the project, and import FirebaseAdapter from that copied file like this:
import { FirebaseAdapter as FirebaseAdapter2 } from "../firebaseAdaptaer2";
And it worked perfectly

In your case you might want to have 2 copies.
If you somehow found a better answer to this please tell me :)
